qutip/random_objects.py

```diff
@@ -150,8 +154,10 @@ def rand_herm(N, density=0.75, dims=None, pos_def=False):
             _check_dims(dims, N, N)
         nvals = N**2*density
         while M.nnz < 0.95*nvals:
-            M = rand_jacobi_rotation(M)
+            M = rand_jacobi_rotation(M, seed=seed)
```

Comment:
Putting the seed here will reseed at each iteration of the while loop.
Better to to it just once at the beginning of the function.